beets/ui/commands.py

```diff
@@ -684,10 +704,18 @@ def choose_match(self, task):
         # Loop until we have a choice.
         candidates, rec = task.candidates, task.rec
         while True:
+            # Gather extra choices from plugins.
+            # TODO: check for conflicts in choices (duplicated letters, etc)
+            # TODO: provide a way to override summary_judment and/or non-timid
+            # mode? Might be useful for some plugins
+            extra_choices = list(chain(*plugins.send('before_choose_candidate',
```

Comment:
The list flattening is to handle extra choices received from various plugins.

As hinted on the TODO, it might be useful to do an extra-pass for solving potential problems with the extra choices  (two plugins picking the same `id` for their choices, the same short letter, etc).